docs/en/operations/settings/query_complexity.md

```diff
@@ -73,6 +73,17 @@ Maximum number of unique keys received from aggregation. This setting lets you l
 What to do when the number of unique keys for aggregation exceeds the limit: 'throw', 'break', or 'any'. By default, throw.
 Using the 'any' value lets you run an approximation of GROUP BY. The quality of this approximation depends on the statistical nature of the data.
 
+## max_bytes_before_external_group_by {#settings-max_bytes_before_external_group_by}
+
+Sets the maximum volume of RAM in bytes that can be used by the [GROUP BY](../../query_language/select.md#select-group-by-clause) operation. The setting is used to avoid the out of memory situations. If memory consumption is bigger than the value of this setting, ClickHouse dumps operation state to the storage disk and continues query processing with much lower performance then in RAM. Memory consumption is calculated at local server even if the query is distributed.
```

Comment:
* is bigger -> exceeds
* "storage disk" sounds weird, maybe just "filesystem"
* lower is just a (possible) consequence, there's still no explanation on how this works
* local server**s** ?